static/js/admin.js

```diff
@@ -408,6 +432,51 @@ exports.setup_page = function () {
             }
         });
     });
+
+    $('.admin_emoji_table').on('click', '.delete', function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var btn = $(this);
+
+        channel.del({
+            url: '/json/realm/emoji/' + encodeURIComponent(btn.attr('data-emoji-name')),
+            error: function (xhr, error_type) {
+                if (xhr.status.toString().charAt(0) === "4") {
+                    btn.closest("td").html(
+                        $("<p>").addClass("text-error").text($.parseJSON(xhr.responseText).msg)
+                    );
+                } else {
+                     btn.text("Failed!");
+                }
+            },
+            success: function () {
+                var row = btn.parents('tr');
+                row.remove();
+            }
+        });
+    });
+
+    $(".administration").on("submit", "form.admin-emoji-form", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var emoji_status = $('#admin-emoji-status');
+        var emoji_table = $('.admin_emoji_table');
+        var emoji = {};
+        $(this).serializeArray().map(function(x){emoji[x.name] = x.value;});
+
+        channel.put({
+            url: "/json/realm/emoji",
+            data: $(this).serialize(),
```

Comment:
`this` corresponds to the form in this context. It usually yields the element you're binding.
An excerpt from https://api.jquery.com/on/:

> When jQuery calls a handler, the this keyword is a reference to the element where the event is being delivered; for directly bound events this is the element where the event was attached and for delegated events this is an element matching selector. (Note that this may not be equal to event.target if the event has bubbled from a descendant element.) To create a jQuery object from the element so that it can be used with jQuery methods, use $( this ).